File: store-java/src/main/java/io/mb/store/controller/DealersController.java
package io.mb.store.controller;

import io.mb.store.resource.dto.DealerVehiclesDto;
import io.mb.store.resource.dto.VehicleDto;
import io.mb.store.service.VehicleService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Flux;

import static io.mb.store.resource.Constants.*;

@RestController
@RequestMapping(DEALERS_MAPPING)
@RequiredArgsConstructor
@Slf4j
public class DealersController {
    private final VehicleService vehicleService;

    @GetMapping
    public Flux<DealerVehiclesDto> getAllVehiclesDetails() {
        log.info("Fetching all vehicles from all dealers");
        Flux<DealerVehiclesDto> vehicles = vehicleService.getAllVehicles();
        log.info("Returned all vehicles available");
        return vehicles;
    }

    @GetMapping(DEALERS_VEHICLES_MAPPING)
    public Flux<VehicleDto> getAllVehiclesFromDealer(
            @PathVariable(DEALER_PARAM) final String dealerId
    ) {
        log.info("Fetching all vehicles available in dealer " + dealerId);
        Flux<VehicleDto> vehicles = vehicleService.getVehiclesFromDealer(dealerId);
        log.info("Returned all vehicles available on dealer " + dealerId);
        return vehicles;
    }
}

File: store-kotlin/src/main/kotlin/io/mb/store/controller/DealersController.kt
package io.mb.store.controller

import io.mb.store.resource.DEALERS_MAPPING
import io.mb.store.resource.DEALER_PARAM
import io.mb.store.resource.DEALER_VEHICLES_MAPPING
import io.mb.store.resource.dto.DealerVehiclesDto
import io.mb.store.resource.dto.VehicleDto
import io.mb.store.service.VehicleService
import kotlinx.coroutines.flow.Flow
import mu.KotlinLogging
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(DEALERS_MAPPING)
class DealersController(
    private val vehicleService: VehicleService,
) {

    private companion object {
        private val logger = KotlinLogging.logger { }
    }

    @GetMapping
    fun getAllVehiclesDetails(): Flow<DealerVehiclesDto> {
        logger.info { "Fetching all vehicles from all dealers" }
        return vehicleService.getAllVehicles().also {
            logger.info { "Returned all vehicles available" }
        }
    }

    @GetMapping(DEALER_VEHICLES_MAPPING)
    fun getAllVehiclesFromDealer(
        @PathVariable(DEALER_PARAM) dealerId: String
    ): Flow<VehicleDto> {
        logger.info { "Fetching all vehicles available in dealer $dealerId" }
        return vehicleService.getVehiclesFromDealer(dealerId).also {
            logger.info { "Returned all vehicles available on dealer $dealerId" }
        }
    }
}